### PR TITLE
[opencv] Move dependency of protobuf from core to dnn/flann

### LINKED ISSUE #13986: [opencv] opencv[core] should not depend on protobuf
****

**Is your feature request related to a problem? Please describe.**
opencv3/opencv4 depends on protobuf in core feature, but which is only used in dnn/flann feature.

protobuf is a library with extremely poor compatibility, it'll terminal your programe at runtime when data differs from library version.

If some projects use specified protobuf version explicitly, then it cannot link with opencv[core], even if it doesn't use dnn feature at all -- otherwise the program will compile and link successfully, but terminate immediate after start up.

**Proposed solution**
We should move denpendency of protobuf from `[core]` to `[flann](opencv3)` or `[dnn](opencv4)`.

**Describe alternatives you've considered**
- For `CONTROL` file:
  - We should move denpendency of protobuf from `[core]` to `[flann](opencv3)` or `[dnn](opencv4)`.
- For `portfile.cmake` file:
  - We should use `={BUILD_opencv_flann}` or `={BUILD_opencv_dnn}` for all build options of protobuf

**Additional context**
None.

### PR DESCRIPTION
**Pull request by ZgblKylin**
**Describe the pull request**
opencv3/opencv4 depends on protobuf in core feature, but which is only used in dnn/flann feature.

protobuf is a library with extremely poor compatibility, it'll terminal your programe at runtime when data differs from library version.

If some projects use specified protobuf version explicitly, then it cannot link with opencv[core], even if it doesn't use dnn feature at all -- otherwise the program will compile and link successfully, but terminate immediate after start up.

- What does your PR fix? 
Fixes #13986

- Which triplets are supported/not supported? Have you updated the CI baseline?
Support all existing triplets.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/opencv4/CONTROL [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Source: opencv4
 Version: 4.3.0
 Port-Version: 1
```
Comment (PhoebeHui) [inline ports/opencv4/CONTROL:3]:
```suggestion
Port-Version: 2
```

File: ports/opencv3/CONTROL [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Source: opencv3
 Version: 3.4.10
```
Comment (PhoebeHui) [inline ports/opencv3/CONTROL:2]:
```suggestion
Version: 3.4.10
Port-Version: 1
```

**Review by PhoebeHui — COMMENTED** [review]
All features for opencv4 test passed with x64-windows. 
Except feature qt and halide, others features for opencv4 test passed with x64-linux, the 2 features does not relate to this PR, so it doesn't block this PR.

**Review by PhoebeHui — APPROVED** [review]

**Comment by ZgblKylin** [comment]
@PhoebeHui Check [microsoft.vcpkg.pr](https://dev.azure.com/vcpkg/public/_build/results?buildId=44067&view=logs&j=a83b7983-e539-5449-9e09-337ff5f52283) failed with osx_x64 saying:

> ##[error]We stopped hearing from agent vcpkg-eg-mac-06. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610

**Comment by PhoebeHui** [comment]
@ZgblKylin, please ignore the failure, the agent machine is broken.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**